gm4_soul_fragments/data/gm4_soul_fragments/functions/soul_fragment/process_check.mcfunction

```diff
@@ -0,0 +1,8 @@
+# @s = abandoned soul fragment
+# at @s
+# from main
+
+# check if there are items or players nearby before processing
+execute if entity @e[type=#gm4_soul_fragments:soul_target,distance=..22] run function gm4_soul_fragments:soul_fragment/process
+
+function gm4_soul_fragments:soul_fragment/age
```

Comment:
This is the only function that calls the function `gm4_soul_fragments:soul_fragment/age`, so those should just be merged. I.e. move the commands from `gm4_soul_fragments:soul_fragment/age` into here instead of calling the function